### ISSUE TITLE
Default to MUMPS if MA57 is not available

### ISSUE THREAD

**Issue by odow** [cvanaret/Uno#32]
This is going to become annoying:
```Julia
julia> model = Model() do
           return AmplNLWriter.Optimizer(UnoSolverCommand())
       end
A JuMP Model
├ solver: AmplNLWriter
├ objective_sense: FEASIBILITY_SENSE
├ num_variables: 0
├ num_constraints: 0
└ Names registered in the model: none

julia> set_attribute(model, "preset", "ipopt")

julia> set_attribute(model, "print_solution", "yes")

julia> @variable(model, x)
x

julia> @objective(model, Min, (x - 1)^2)
x² - 2 x + 1

julia> optimize!(model)
libc++abi: terminating due to uncaught exception of type std::invalid_argument: Linear solver name is unknown
```
The solution is
```Julia
julia> set_attribute(model, "linear_solver", "MUMPS")

julia> optimize!(model)
Problem /var/folders/bg/dzq_hhvx1dxgy6gb5510pxj80000gn/T/jl_VfYlG8/model.nl_scaled_equalityconstrained_boundrelaxed
1 variables, 0 constraints

Options:
... etc
```

But better options would be to:

 * default to MUMPS if MA57 is not available and MUMPS is
 * improve the error message to print the name of the available solvers and to tell the user to set the `linear_solver` option

**Comment by cvanaret** [comment]
I will change the default solvers based on what's available.

**Comment by odow** [comment]
Maybe the same with the presets? If we haven't compiled BQPD then make `-preset ipopt` the default?

**Comment by cvanaret** [comment]
Quick question: you used the command `set_attribute(model, "preset", "ipopt")` in your code. This should be an attribute of the solver/optimizer, rather than of the model.
Would it be possible to have something like:
 ```
solver = AmplNLWriter.Optimizer(UnoSolverCommand())
set_attribute(solver, "preset", "ipopt")
...
```
?

**Comment by odow** [comment]
There are multiple ways that you can pass options to a solver: https://jump.dev/JuMP.jl/stable/manual/models/#solver_options

None of this needs to concern Uno though. We'll pack them into the appropriate place in the CLI.

**Comment by cvanaret** [comment]
Setting the default subproblem solvers + improving the error messages was addressed in commit d8571b6.
I will address the default preset in a separate issue.

**Comment by odow** [comment]
Great. I confirm that this worked in the latest build of Uno_jll
